zeek/plugin/src/runtime-support.cc

```diff
@@ -86,9 +86,15 @@ void rt::raise_event(EventHandlerPtr handler, const hilti::rt::Vector<Val*>& arg
     ::mgr.QueueEventFast(handler, vl);
 }
 
-BroType* rt::event_arg_type(EventHandlerPtr handler, uint64_t idx) {
+BroType* rt::event_arg_type(EventHandlerPtr handler, uint64_t idx, std::string_view location) {
     assert(handler);
-    return (*handler->FType()->ArgTypes()->Types())[idx];
+
+    auto zeek_args = handler->FType()->ArgTypes()->Types();
+    if ( idx >= zeek_args->length() )
```

Comment:
That's a comparison between signed and unsigned variables. Let's cast on of them to fix the semantics we want. Since `zeek_args->length()` is signed we also need to fix the format string below.

Reply:
Good catch, will fix.